src/Microsoft.AspNet.Mvc.Razor/TagHelpers/UrlResolutionTagHelper.cs

```diff
@@ -0,0 +1,206 @@
+    [TargetElement("img", Attributes = "src")]
+    [TargetElement("script", Attributes = "src")]
+    [TargetElement("source", Attributes = "src")]
+    [TargetElement("track", Attributes = "src")]
+    [TargetElement("input", Attributes = "src")]
+    [TargetElement("input", Attributes = "formaction")]
+    [TargetElement("button", Attributes = "formaction")]
+    [TargetElement("form", Attributes = "action")]
+    [TargetElement("blockquote", Attributes = "cite")]
+    [TargetElement("del", Attributes = "cite")]
+    [TargetElement("ins", Attributes = "cite")]
+    [TargetElement("q", Attributes = "cite")]
+    [TargetElement("menuitem", Attributes = "icon")]
+    [TargetElement("html", Attributes = "manifest")]
+    [TargetElement("object", Attributes = "data")]
+    [TargetElement("object", Attributes = "archive")]
+    [TargetElement("applet", Attributes = "archive")]
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    public class UrlResolutionTagHelper : TagHelper
+    {
+        private static readonly IReadOnlyDictionary<string, IEnumerable<string>> ElementAttributeLookups =
+            new Dictionary<string, IEnumerable<string>>(StringComparer.OrdinalIgnoreCase)
+            {
+                { "a", new[] { "href" } },
+                { "area", new[] { "href" } },
+                { "link", new[] { "href" } },
+                { "base", new[] { "href" } },
+                { "video", new[] { "poster", "src" } },
+                { "audio", new[] { "src" } },
+                { "embed", new[] { "src" } },
+                { "iframe", new[] { "src" } },
+                { "img", new[] { "src" } },
+                { "script", new[] { "src" } },
+                { "source", new[] { "src" } },
+                { "track", new[] { "src" } },
+                { "input", new[] { "src", "formaction" } },
+                { "button", new[] { "formaction" } },
+                { "form", new[] { "action" } },
+                { "blockquote", new[] { "cite" } },
+                { "del", new[] { "cite" } },
+                { "ins", new[] { "cite" } },
+                { "q", new[] { "cite" } },
+                { "menuitem", new[] { "icon" } },
+                { "html", new[] { "manifest" } },
+                { "object", new[] { "data", "archive" } },
+                { "applet", new[] { "archive" } },
+            };
+
+        /// <summary>
+        /// Creates a new <see cref="UrlResolutionTagHelper"/>.
+        /// </summary>
+        /// <param name="urlHelper">The <see cref="IUrlHelper"/>.</param>
+        /// <param name="htmlEncoder">The <see cref="IHtmlEncoder"/>.</param>
+        public UrlResolutionTagHelper(IUrlHelper urlHelper, IHtmlEncoder htmlEncoder)
+        {
+            UrlHelper = urlHelper;
+            HtmlEncoder = htmlEncoder;
+        }
+
+        /// <inheritdoc />
+        public override int Order
+        {
+            get
+            {
+                return int.MinValue;
```

Comment:
Understand we want this one to run early but earliest possible means the only thing anyone can do to override this helper is to remove it. Doesn't cover for example changing a specific attribute this helper targets. Further comments in `DefaultOrder` imply we shouldn't be using `0` as we do today in other tag helpers or `MinValue`: `// ... framework code that depends on order should be ordered between -1 to -1999.`.

Suggest using MVC-traditional `DefaultOrder.DefaultFrameworkSortOrder` (`-1000`). If instead we decide the rest of our tag helpers should have `DefaultOrder.DefaultFrameworkSortOrder`, then this one should have `DefaultOrder.DefaultFrameworkSortOrder - 999`.